### ISSUE TITLE
Keras does not save weights properly

### ISSUE THREAD

**Issue by gcelano** [keras-team/keras#19693]
I am using Tensorflow 2.15.1 and, if I use model.save_weights("./dir/mymodel.bin") and subsequently try to load the weights with model.load_weights (in a different process), the model does not return the expected results (when using model.evaluate or .predict).  My model has a custom loss function and metric, but this should not be a problem, since I try to load weights only. Any idea why I get this error?

**Comment by sachinprasadhs** [comment]
Could you please try with the Keras 3 version and let us know your observations and possibly some reproducible code. Thanks

**Comment by gcelano** [comment]
I installed Tensorflow 2.16.1, but the it cannot detect the GPUs.

**Comment by gcelano** [comment]
The problem is also with Tensorflow 2.16. Please, run the code here:

https://git.informatik.uni-leipzig.de/celano/dithrax_test

**Comment by sonali-kumari1** [comment]
Hi @gcelano -

Could you please provide a standalone code in the form of a colab notebook to reproduce this issue? Thanks!
